### DISCUSSION TITLE
Setting Up NVIDIA GPU for TensorFlow on WSL2: A quick Guide

### DISCUSSION THREAD

**Discussion by giscus** [PradeepLoganathan/pradeepl-blog#99] [BlogPosts]
# Setting Up NVIDIA GPU for TensorFlow on WSL2: A quick Guide

Setting Up NVIDIA GPU for TensorFlow on WSL2: A quick Guide

https://pradeepl.com/blog/installing-nvidia-cuda-tensorflow-on-windows-wsl2/

**Comment by Acgholic** [comment]
I installed cuda 12.5 using  NVIDIA’s Package Repository with deb(local) and I can't find the path such as /usr/local/cuda/include and /usr/local/cuda/lib64, but I can find path such as /usr/local/cuda-12.5/bin etc. My question is should I do sudo cp cuda/include/cudnn*.h /usr/local/cuda/include, when the path such as include was not existing?

**Reply by Acgholic** [reply]
When I can't find path such as /usr/local/cuda/include, I can find \usr\local\cuda-12.5\targets\x86_64-linux\include and \usr\local\cuda-12.5\targets\x86_64-linux\lib. Are the two paths corresponding to the /usr/local/cuda/include and /usr/local/cuda/lib64?

**Comment by hugin78** [comment]
Hello
Unfortunately, I have spent a lot of time trying to get your setup to work without success. I ended up learning that you may have overlooked that tensorflow actually supports installing cuda and cuDNN out of the box. You just have to follow these steps after installing the driver:

wsl --install
sudo apt update
sudo add-apt-repository ppa:deadsnakes/ppa
sudo apt install python3.12 python3.12-venv  python3-pip
python3 -m venv mygpuenv
source mygpuenv/bin/activate
pip install tensorflow[and-cuda]
python3 -c "import tensorflow as tf; print(tf.config.list_physical_devices('GPU'))"

**Comment by MathProf1234** [comment]
Hi, 

Following the instructions I get stuck at  
sudo apt-get -y install cuda
.....
> The following packages have unmet dependencies:
>  libcufile-11-4 : Depends: liburcu6 but it is not installable
>  xserver-xorg-video-nvidia-470 : Depends: xorg-video-abi-24 but it is not installable or
....

What do I do?

Thanks,

**Comment by PradeepLoganathan** [comment]
Can you try use the -f flag: Use apt install -f to tell APT to find and install missing packages.

**Comment by MathProf1234** [comment]
Thanks, so instead of
sudo apt-get -y install cuda
do I write
sudo apt-get -y -f install cuda
?

Also at https://developer.download.nvidia.com/compute/cuda/repos
there is wsl-ubuntu directory. Could that be better than ubuntu2004 when
using ubuntu under WSL2?

Thanks
